scipy/special/_input_validation.py

```diff
@@ -0,0 +1,19 @@
+import operator
+
+from numpy import floor
+
+
+def _nonneg_int_or_fail(n, var_name, strict=True):
+    try:
+        if strict:
+            # Raises an exception if float
+            n = operator.index(n)
+        elif n == floor(n):
```

Comment:
Yeah, actually I don't know we aren't just using math.floor here. We just inherited `_nonneg_int_or_fail` from somewhere, but it will never be an array argument. Will use math.floor.

Reply:
Yeah, this was written in the early days and was in `_basic.py`. We just separated it out to avoid a circular dependency.

Reply:
Have done this.